### ISSUE TITLE
Configuration to disable/enable of the REST API

### ISSUE THREAD

**Issue by adeelx** [0xERR0R/blocky#1109]
I have currently deployed blocky with the following port configuration:
```
ports:
  dns: 53
  tls: 853
  https: 443
```
The [interfaces documentation](https://0xerr0r.github.io/blocky/v0.21/interfaces/) indicates this:
```
If http listener is enabled, blocky provides REST API.
```
Which made me think that disabling `4000` port (http) would also block access to the management API.

But it seems like I'm still able to access the API over `https`.

Given that `https` is required for DNS-over-HTTPS I'm at a loss on how I'd go about blocking access to this management API since I don't want anonymous access to this API.

I could go ahead and put blocky behind a Caddy server and block the management API from there but I was wondering if there's something I'm missing here.

**Comment by 0xERR0R** [comment]
Yes, you're right. If http or https listener is enabled, all http based interfaces (REST, prometheus if enabled and DoH) are available. For your scenario, it would be probably better to disable REST-API at all? If you need REST-Api, I would recommend to configure reverse proxy with authentication support for this url.

**Comment by adeelx** [comment]
Indeed, I'm okay with disabling the REST-API completely, but it doesn't seem like that's possible, is it?
Ideally, I'd like to disable the Rest API while maintaining the DNS-over-HTTPS functionality.

**Comment by 0xERR0R** [comment]
I think, we can introduce new configuration block like prometheus:
```yaml
api:
   enable: true
```

maybe we can enhance it later with other config options (like basic auth, ...)

**Comment by adeelx** [comment]
Agreed, that would be the best course of action. Thank you for taking an enhancement @0xERR0R

**Comment by 0xERR0R** [comment]
will be fixed via #1206

**Comment by michivonah** [comment]
For security reasons, I would also like to disable the REST API or at least secure it with authentication. What is the status here? Can we expect this to happen in the next few months?

Thanks a lot for providing us with this tool.
